TF_Service_and_Tooling_Unity/Assets/TouchFree/ServiceShared/Scripts/Configuration/ConfigFiles/ConfigFile.cs

```diff
@@ -81,7 +81,14 @@ protected TData LoadConfig_Internal()
 
         protected void SaveConfig_Internal(TData config)
         {
-            File.WriteAllText(_ConfigFilePath, JsonUtility.ToJson(config, true));
+            try
+            {
+                File.WriteAllText(_ConfigFilePath, JsonUtility.ToJson(config, true));
+            }
+            catch
+            {
+                PermissionController.hasFilePermission = false;
```

Comment:
Because this only gets set when either creating a default config file, or saving a config file, does that mean that the popup won't appear until users try (and the app fails) to make a change or does something call save immediately and reveal the blockers? (I didn't have an easy way to test this on hand w/  my permissions chaos)

Reply:
I added the new PermissionController.cs which does exactly that, it asks a config to try and save (which will load first because we access a config)